pkg/controller/datavolume/snapshot-clone-controller.go

```diff
@@ -130,11 +130,6 @@ func (r *SnapshotCloneReconciler) prepare(syncState *dvSyncState) error {
 	if err := r.populateSourceIfSourceRef(dv); err != nil {
 		return err
 	}
-	if dv.Status.Phase == cdiv1.Succeeded {
-		if err := r.cleanup(syncState); err != nil {
-			return err
-		}
-	}
```

Comment:
surprised to see changes to this controller and pvc clone controller.  Please elaborate.  Concerrned about the removal of calls to cleanup()

Reply:
you can read the commit message that did this change, it explains why. if not clear after that let me know

Reply:
It's not the end of the world, but I think it would be better if the callback functions were dumb.  They just do a thing.  and the framework makes sure it is called at the right time.  Maybe there can be a `done` callback that is called when the DV is succeeded and `cleanup` when it's deleted?

Reply:
I think how it was before that the prepare called cleanup was yuck.
I think in the cleanup you do different cleanups according to different conditions, so I think the fact that you can decide in the cleanup callback itself when and what should be cleaned is good.. don't think that adding another done callback that will be called when dv is succeeded and will do the exact same thing as the cleanup is valuable.